### PR TITLE
Change: make crash chance at short runway independent of plane crash …

### PR DESCRIPTION
**Pull request by Eddi-z**
…rate setting (there's a cheat for this)

### CODE REVIEW

**Comment by SamuXarick** [comment]
```
plane crash | large aircraft | small airport | cheat enabled | probability
------------+----------------+---------------+---------------+------------------------
          0 |             no |            no |            no | (return)
          0 |             no |            no |           yes | (return)
          0 |             no |           yes |            no | (return)
          0 |             no |           yes |           yes | (return)
          0 |            yes |            no |            no | (return)
          0 |            yes |            no |           yes | (return)
          0 |            yes |           yes |            no | 3276
          0 |            yes |           yes |           yes | (return)
------------+----------------+---------------+---------------+------------------------
          1 |             no |            no |            no | 0x4000 << 1 / 1500 = 21
          1 |             no |            no |           yes | 0x4000 << 1 / 1500 = 21
          1 |             no |           yes |            no | 0x4000 << 1 / 1500 = 21
          1 |             no |           yes |           yes | 0x4000 << 1 / 1500 = 21
          1 |            yes |            no |            no | 0x4000 << 1 / 1500 = 21
          1 |            yes |            no |           yes | 0x4000 << 1 / 1500 = 21
          1 |            yes |           yes |            no | 3276
          1 |            yes |           yes |           yes | 0x4000 << 1 / 1500 = 21
------------+----------------+---------------+---------------+------------------------
          2 |             no |            no |            no | 0x4000 << 2 / 1500 = 43
          2 |             no |            no |           yes | 0x4000 << 2 / 1500 = 43
          2 |             no |           yes |            no | 0x4000 << 2 / 1500 = 43
          2 |             no |           yes |           yes | 0x4000 << 2 / 1500 = 43
          2 |            yes |            no |            no | 0x4000 << 2 / 1500 = 43
          2 |            yes |            no |           yes | 0x4000 << 2 / 1500 = 43
          2 |            yes |           yes |            no | 3276
          2 |            yes |           yes |           yes | 0x4000 << 2 / 1500 = 43
```

**Comment by Eddi-z** [comment]
Note: this is meant as an alternate approach to #7293

**Comment by Eddi-z** [comment]
A quick rundown of how planecrash chances work:
since da3ff51759f75ef76b896d19a5ca6b88a586e724 the plane crash chance is rerolled every tick that the plane is on the runway, and its speed is larger than the taxiing speed
since fa01b25f740d92ffa74de4bc630ca42cdc8046a0 the plane crash chance is affected by a setting to reduce (halve) or eliminate the crash chance

thus the chance for a plane crash on landing is `(value of above table) * (time to decelerate from approach speed to taxiing speed) / 2^22` (giving a slight bonus to slow planes)

**Comment by Eddi-z** [comment]
comparison between crash chances
```
set-|  airport  |      crash chance 
ting|  size     | master | #7293 | #7302
----+-----------+--------+-------+-------
  0 | ok        |      0 |     0 |     0
    | too small |      0 |     0 |  3276
    | cheated   |      0 |     0 |     0
----+-----------+--------+-------+-------
  * | ok        |      - |     0 |     -
    | too small |      - |  1638 |     -
    | cheated   |      - |     0 |     -
----+-----------+--------+-------+-------
  1 | ok        |     21 |    21 |    21
    | too small |   1638 |  1638 |  3276
    | cheated   |     21 |    21 |    21
----+-----------+--------+-------+-------
  2 | ok        |     43 |    43 |    43
    | too small |   3276 |  3276 |  3276
    | cheated   |     43 |    43 |    43
```

**Comment by SamuXarick** [comment]
The cheated value is 21, not 0

**Comment by TrueBrain** [comment]
I very much like this approach above adding "yet another setting". We have a cheat to prevent it, and it is fully what I expected the game already does. So yeah, lets do this one! :)

**Comment by SamuXarick** [comment]
This feels wrong. 
You expect crashes when you set plane crashes to "none"? Even if you rename this better, it's still going to feel as if an option is missing. None means none!

**Comment by PeterN** [comment]
Yeah, "plane crashes" are a disaster setting that the player has no control over. Sending a large plane to a small airport is a user error.

**Comment by SamuXarick** [comment]
> Yeah, "plane crashes" are a disaster setting that the player has no control over. Sending a large plane to a small airport is a user error.

It is only a user error if this patch is applied, because it currently isn't on master. And I don't believe people see it as a bug. There are some AIs that check whether this setting is none to decide whether to buy large airplanes on short runways. 
This sudden change of setting behaviour baffles me… From a moment to another you decide a feature that is working fine is a "bug" and then you "fix" it (break it, change its meaning, etc...).

**Comment by nielsmh** [comment]
It definitely feels exploit-y to allow large planes to land at small airports at any time.

**Comment by TrueBrain** [comment]
@SamuXarick : I suggest you take a few to cool off. You are being a bully at the moment, because you are not getting what you set out to get. This is not acceptable behavior (even more in combination how you react on IRC). Please cool down before replying. It really helps to keep conversations productive.

That said, we have said enough about this. Without new argumentation, this is final: we consider large planes on small airports user error, which results in a plane crash. This can be avoided with a cheat, just as you would expect based on the cheat description.

**Comment by Berbe** [comment]
@SamuXarick I suppose nothing has really ever been *defined* before, hence the confusion.
The thing is no normal user ever had a chance to understand how that feature *really* worked, without looking at the code.

Making the label on the setting explaining best what really happens could help finally putting everyone of the same page?

File: src/lang/english.txt [hunk 1]
```diff
@@ -1257,7 +1257,7 @@ STR_CONFIG_SETTING_PLANE_SPEED                                  :Plane speed fac
 STR_CONFIG_SETTING_PLANE_SPEED_HELPTEXT                         :Set the relative speed of planes compared to other vehicle types, to reduce the amount of income of transport by aircraft
 STR_CONFIG_SETTING_PLANE_SPEED_VALUE                            :1 / {COMMA}
 STR_CONFIG_SETTING_PLANE_CRASHES                                :Number of plane crashes: {STRING2}
-STR_CONFIG_SETTING_PLANE_CRASHES_HELPTEXT                       :Set the chance of an aircraft crash happening
+STR_CONFIG_SETTING_PLANE_CRASHES_HELPTEXT                       :Set the chance of a random aircraft crash happening.{}Does not affect plane crashes due to too short runways.
```
Comment (LordAro) [inline src/lang/english.txt]: "to too", while correct, reads weirdly. Perhaps "Does not affect plane crashes on runways that are too short [for the plane]" ?

Comment (Berbe) [reply src/lang/english.txt]: > Does not affect plane crashes on too short runways.

?

**Comment by LordAro** [comment]
For what it's worth, I do agree that "None" while still expecting crashes (even if it is users that screw up) is an issue - but that's a wording thing, rather than anything that needs changing functionally

**Comment by Eddi-z** [comment]
> And I don't believe people see it as a bug.

You keep saying that, but i've never seen any of these hypothetical people. The overwhelming response (at least on this side of the bubble) has been: "Wait, i thought that is how it worked already"

File: src/lang/english.txt (hunk 1 above)
Comment (nielsmh) [reply src/lang/english.txt]: >Large airplanes always have a risk of crashing when landing on small airports.

**Review by TrueBrain — APPROVED** [review]

**Pull request merged**